src/quicktube.js

```diff
@@ -0,0 +1,284 @@
+            this.hidePosterFrame();
+            this.addActiveState();
+            window.dispatchEvent(new Event('quicktube:play'));
+        }
+    }
+
+    addActiveState() {
+        this.videoEl.classList.add(this.options.activeClass);
+        this.videoEl.classList.remove(this.options.pausedClass);
+    }
+
+    removeActiveState() {
+        this.videoEl.classList.remove(this.options.activeClass);
+        this.videoEl.classList.add(this.options.pausedClass);
+        this.videoEl.removeAttribute('data-video-playing');
+    }
+
+    stopVideo() {
+        if (!this.quicktubePlayer) {
+            return;
+        }
+
+        this.quicktubePlayer.stopVideo();
+        this.removeActiveState();
+        this.showPosterFrame();
+        window.dispatchEvent(new Event('quicktube:stop'));
+    }
+
+    hidePosterFrame() {
+        this.poster.classList.add(this.options.posterFrameHiddenClass);
+    }
+
+    showPosterFrame() {
+        this.poster.classList.remove(this.options.posterFrameHiddenClass);
+    }
+
+    createIframePlayer() {
+        const iframe = document.createElement('iframe');
+        iframe.src = this._domain + this.videoId + this._settings;
+        iframe.width = '100%';
+        iframe.id = this.videoId;
+        iframe.className = this.iframeClass;
+        return iframe;
+    }
+
+
+    onPlayerReady(event) {
+        const isPlaying = this.videoEl.getAttribute('data-video-playing');
+        if (!this.isMobileSafari) {
+            if (isPlaying) {
+                // TODO evaluate if this is needed
+                // Not sure it ever gets to this point
+                this.stopVideo();
+            } else {
+                this.videoEl.setAttribute('data-video-playing', true);
+                event.target.playVideo();
+            }
+        }
+    }
+
+    // listen for play, pause and end states
+    // also report % played every second
+    onPlayerStateChange(event) {
+        if (event.data === YT.PlayerState.PLAYING) {
+            this.videoEl.setAttribute('data-video-playing', true);
+            this.addActiveState();
+            setTimeout(this.onPlayerPercent.bind(this), 1000, event.target);
+        }
+
+        if (event.data === YT.PlayerState.PAUSED) {
+            this.removeActiveState();
+        }
+
+        const videoData = event.target.getVideoData();
+        let label = videoData.title;
+        // Get title of the current page
+        const pageTitle = document.title;
+
+        // TODO figure out what this is all doing and test it
+        if (this.options.trackAnalytics) {
+            if (event.data === YT.PlayerState.PLAYING && YT.gaLastAction === 'p') {
+                label = `Video Played - ${videoData.title}`;
+                trackEvent({
+                    event: 'youtube',
+                    eventCategory: 'Youtube Videos',
+                    eventAction: pageTitle,
+                    eventLabel: label,
```

Comment:
Lots of the events on the page have the same `event`, `eventCategory` and `eventAction`, so this should be extracted to a separate method.